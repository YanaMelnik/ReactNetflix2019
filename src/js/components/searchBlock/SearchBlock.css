.search-film__block {
    min-height: 170px;
    background: linear-gradient(
            rgba(0, 0, 0, 0.8),
            rgba(0, 0, 0, 0.49)),
    url(https://xage.ru/media/posts/2017/5/23/netflix-viewing-pattern-comedy-at-breakfast.jpg
    );
}
