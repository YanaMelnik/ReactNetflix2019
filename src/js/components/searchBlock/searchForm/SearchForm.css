.search-film__name-title {
    display: block;
    text-transform: uppercase;
    color: white;
    font-family: Arial, serif;
    font-weight: 300;
    margin: 15px 0 20px;
}

.search-film__name-input {
    display: block;
    width: 96%;
    border: 0 solid;
    border-bottom: 2px solid #f34c70;
    background: black;
    height: 30px;
    color: white;
    font-size: 14px;
    padding-left: 20px;
}

.search-by__title {
    color: white;
    text-transform: uppercase;
    font-family: Arial, serif;
    font-size: 12px;
    margin-right: 20px;
    font-weight: 600;
}

.search-by__input {
    display: none;
}

.search-by__label {
    display: inline-block;
    background-color: #4c4a4a;
    padding: 2px 15px;
    font-family: Arial, serif;
    font-size: 12px;
    border-radius: 4px;
    color: white;
    font-weight: 400;
    margin: 10px 10px 30px;
    text-transform: uppercase;
}

.search-by input[type="radio"]:checked + label {
    background-color: #f34c70;
}

.search-by__label:hover {
    background-color: #ffc1c9;
}

.search-by__action-block {
    display: flex;
    justify-content: space-between;
}
