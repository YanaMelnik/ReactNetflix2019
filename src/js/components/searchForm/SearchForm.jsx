import React from 'react';
import Logo from '../common/logo/Logo.jsx';
import Button from '../common/button/Button.jsx';

const SearchForm = () => <div>
    <Logo/>
    <form>
      <label
        htmlFor='search-film__name'
        className='search-film__name-title'>
        Find your movie
      </label>

      <input
        id='search-film__name'
        className='search-film__name-input'/>

      <span className='search__by'>
            <span>Seach by</span>
            <span>
              <input name='search__by' type='radio' value='title' checked/>
              Title
            </span>
            <span>
              <input name='search__by' type='radio' value='gender'/>
              Gender
            </span>
          </span>
      <Button name='Search' action='' sizeBig={true}/>
    </form>
  </div>;

export default SearchForm;
