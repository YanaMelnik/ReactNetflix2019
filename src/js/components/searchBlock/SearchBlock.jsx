import React, { Component } from 'react';
import PropTypes from 'prop-types';
import SearchForm from './searchForm/SearchForm';

import './SearchBlock.css';

class SearchBlock extends Component {
  render() {
    return <div className='search-film__block'>
      <SearchForm searchBy={this.props.searchBy}/>
    </div>;
  }
}

SearchBlock.propTypes = {
  searchBy: PropTypes.string.isRequired,
};

export default SearchBlock;
