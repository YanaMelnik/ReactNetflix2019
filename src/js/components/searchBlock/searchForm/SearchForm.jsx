import React from 'react';
import Logo from '../../common/logo/Logo';
import Button from '../../common/button/Button';

import './SearchForm.css';

const SearchForm = () => <div>
    <Logo/>
    <form className='wrapper'>
      <label
        htmlFor='search-film__name'
        className='search-film__name-title'>
        Find your movie
      </label>

      <input
        id='search-film__name'
        className='search-film__name-input'/>

      <div className='search-by__action-block'>
        <span className='search-by'>
        <span className='search-by__title'>Seach by</span>
        <input className='search-by__input' id='search-by__title' name='search-by' type='radio' value='title' checked/>
        <label className='search-by__label' htmlFor='search-by__title'>
          Title
        </label>

        <input className='search-by__input' id='search-by__gender' name='search-by' type='radio' value='gender'/>
        <label className='search-by__label' htmlFor='search-by__gender'>
          Gender
        </label>
        </span>
        <Button text='Search' action=''/>
      </div>
    </form>
  </div>;

export default SearchForm;
