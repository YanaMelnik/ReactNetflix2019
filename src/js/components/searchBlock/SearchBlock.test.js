import React from 'react';
import { shallow } from 'enzyme';
import SearchBlock from './SearchBlock';
import SearchForm from './searchForm/SearchForm';

describe('SearchBlock', () => {
    let component;
    beforeEach(() => {
        component = shallow(<SearchBlock/>);
    });
    it('should render component correctly in case when all props defined', () => {
        expect(component).toMatchSnapshot();
    });

    it('should contain search form', () => {
        expect(component.find(SearchForm)).toHaveLength(1);
    });

    it('should show search block with correct classes', () => {
        expect(component.find('div.search-film__block')).toHaveLength(1);
    });
});
