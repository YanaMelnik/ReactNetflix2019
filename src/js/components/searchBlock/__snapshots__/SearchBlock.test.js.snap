// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBlock should render component correctly in case when all props defined 1`] = `
<div
  className="search-film__block"
>
  <SearchForm />
</div>
`;
